core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -13,543 +13,593 @@
+      return this.getQuestionContributionsSummary(suggestionIdToSuggestions);
+    }
+  }
+
+  getActiveDropdownTabChoice(): string {
+    if (this.activeTabType === this.TAB_TYPE_REVIEWS) {
+      if (this.activeSuggestionType === this.SUGGESTION_TYPE_QUESTION) {
+        return 'Review Questions';
+      }
+      return 'Review Translations';
+    }
 
-      ctrl.onClickBackToReviewableLessons = function() {
-        ctrl.activeExplorationId = null;
-      };
+    if (this.activeSuggestionType === this.SUGGESTION_TYPE_QUESTION) {
+      return 'Questions';
+    }
 
-      ctrl.loadOpportunities = function() {
-        return ctrl.loadContributions(/* Param shouldResetOffset= */ true);
-      };
+    return 'Translations';
+  }
+
+  switchToTab(tabType: string, suggestionType: string): void {
+    this.activeSuggestionType = suggestionType;
+    this.activeTabType = tabType;
+    this.contributionAndReviewService.setActiveTabType(tabType);
+    this.contributionAndReviewService.setActiveSuggestionType(suggestionType);
+    this.activeDropdownTabChoice = this.getActiveDropdownTabChoice();
+    this.dropdownShown = false;
+    this.activeExplorationId = null;
+    this.contributionOpportunitiesService
+      .reloadOpportunitiesEventEmitter.emit();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  loadReviewableTranslationOpportunities(): Promise<GetOpportunitiesResponse> {
+    return this.contributionOpportunitiesService
+      .getReviewableTranslationOpportunitiesAsync(
+        this.translationTopicService.getActiveTopicName())
+      .then((response) => {
+        const opportunitiesDicts = [];
+        response.opportunities.forEach(opportunity => {
+          const opportunityDict = {
+            id: opportunity.getExplorationId(),
+            heading: opportunity.getOpportunityHeading(),
+            subheading: opportunity.getOpportunitySubheading(),
+            actionButtonTitle: 'Translations'
+          };
+          opportunitiesDicts.push(opportunityDict);
+        });
 
-      ctrl.loadMoreOpportunities = function() {
-        return ctrl.loadContributions(/* Param shouldResetOffset= */ false);
+        return {
+          opportunitiesDicts: opportunitiesDicts,
+          more: response.more
+        };
+      });
+  }
+
+  onClickReviewableTranslations(explorationId: string): void {
+    this.activeExplorationId = explorationId;
+  }
+
+  onClickBackToReviewableLessons(): void {
+    this.activeExplorationId = null;
+  }
+
+  loadContributions(shouldResetOffset: boolean):
+    Promise<GetOpportunitiesResponse> {
+    if (!this.activeTabType || !this.activeSuggestionType) {
+      return new Promise((resolve, reject) => {
+        resolve({opportunitiesDicts: [], more: false});
+      });
+    }
+    const fetchFunction = this.tabNameToOpportunityFetchFunction[
+      this.activeSuggestionType][this.activeTabType];
+
+    this.contributions = {};
```

Comment:
@heyimShivam I believe this line should be moved to `ngOnInit`. The original `loadContributions` function did not have this line. On develop, this is causing errors when trying to open a suggestion review modal, i.e. clicking the "Review" button below logs a console error and does not do anything.

<img width="1727" alt="Screen Shot 2022-08-05 at 5 08 58 PM" src="https://user-images.githubusercontent.com/9004726/183186544-94010dee-a035-4b14-bb83-73e943ec7935.png">

Reply:
Will fix this in https://github.com/oppia/oppia/pull/15851.